io-backend/src/main/java/agh/io/iobackend/controller/GameRoomController.java

```diff
@@ -150,8 +151,9 @@ public void leaveGameRoom(@PathVariable Long id, @PathVariable Long user) throws
         if (gameRoom.getGameStarted()){
             gameService.removeFromGame(id, user);
         }
-        if (gameRoom.getUserList().size() == 0) {
+        if (gameRoom.getUserList().size() == 0 || Objects.equals(gameRoom.getGameMasterID(), user)){
             try {
+                gameService.endGame(gameRoom.getGame().getGameId());
```

Comment:
To chyba powoduje, że jak wychodzę z pokoju to dostaję taki błąd na backendzie:
Cannot invoke "agh.io.iobackend.model.game.Game.getGameId()" because the return value of "agh.io.iobackend.model.game.GameRoom.getGame()" is null

i nie mogę wyjść z pokoju

Reply:
No tak, to było źle napisane, ale i tak nadal są błędy, bo mimo że ten koleś wyszedł, to nadal pyta o users-list i game-started, chyba że to zmieniłeś w swojej wersji - jeśli tak, wrzuciłbyś te zmiany na tego brancha moze? 

Bo w sumie teraz tak sobie myślę, że jak GameMaster wychodzi, to ja od razu zamykam grę (i pokój też) itd no i pozostali gracze będą pytali o grę która nie istnieje - więc może dodać jakiś status END i stąd będą wiedzieli, że to już koniec. Może to jest trochę naokoło, ale na razie, nie wiem, jak inaczej.

Reply:
Albo jak na froncie jest jakos zapisane kto jest GameMasterem, to można sprawdzać, czy on jest w grze/pokoju - jak nie, no to gry tez juz nie ma i trzeba przestać o nią pytać serwer. Nie wiem, czy to ma szansę zadziałać i co nam jest łatwiej zgrać.

Reply:
> No tak, to było źle napisane, ale i tak nadal są błędy, bo mimo że ten koleś wyszedł, to nadal pyta o users-list i game-started, chyba że to zmieniłeś w swojej wersji - jeśli tak, wrzuciłbyś te zmiany na tego brancha moze?
> 

Jeśli chodzi ci o sam pokój to działa dobrze teraz. Ten co wyszedł nie pyta - pyta ten co został